nflow-engine/src/main/java/io/nflow/engine/NflowEngine.java

```diff
@@ -0,0 +1,180 @@
+ *
+ * Use this if you want to start just the nflow-engine.
+ */
+public class NflowEngine implements AutoCloseable {
+
+    private final AnnotationConfigApplicationContext ctx;
+    private final WorkflowLifecycle workflowLifecycle;
+
+    private final ArchiveService archiveService;
+    private final HealthCheckService healthCheckService;
+    private final StatisticsService statisticsService;
+    private final WorkflowDefinitionService workflowDefinitionService;
+    private final WorkflowInstanceService workflowInstanceService;
+    private final WorkflowExecutorService workflowExecutorService;
+
+    /**
+     * Starts up the NflowEngine with WorkflowDispatcher running in a thread.
+     * Property nflow.autostart controls if the thread is started automatically.
+     * If nflow.autostart=false, then the thread can be started with start() method.
+     * @param dataSource
+     * @param sqlVariants
+     * @param workflowDefinitions
+     */
+    public NflowEngine(DataSource dataSource,
+                       SQLVariants sqlVariants,
+                       Collection<AbstractWorkflowDefinition<? extends WorkflowState>> workflowDefinitions) {
+        ctx = new AnnotationConfigApplicationContext();
+
+        ctx.registerBean("nflowDatasource", DataSource.class, () -> dataSource);
+        ctx.registerBean(SQLVariants.class, () -> sqlVariants);
+        ctx.register(EngineConfiguration.class, NflowEngineSpringConfig.class);
+        ctx.refresh();
+
+        workflowLifecycle = ctx.getBean(WorkflowLifecycle.class);
+
+        workflowDefinitionService = ctx.getBean(WorkflowDefinitionService.class);
+        workflowDefinitionService.setWorkflowDefinitions(workflowDefinitions);
+
+        archiveService = ctx.getBean(ArchiveService.class);
+        healthCheckService = ctx.getBean(HealthCheckService.class);
+        statisticsService = ctx.getBean(StatisticsService.class);
+        workflowInstanceService = ctx.getBean(WorkflowInstanceService.class);
+        workflowExecutorService = ctx.getBean(WorkflowExecutorService.class);
+    }
+
+    /**
+     * For manually starting dispatcher thread.
+     * This starts the nflow-engine if property nflow.autostart=false.
+     */
+    public void start() {
+        workflowLifecycle.start();
+    }
+
+    /**
+     * Pauses a running nFlow engine.
+     */
+    public void pause() {
+        workflowLifecycle.pause();
+    }
+
+    /**
+     * Resumes paused the nFlow engine.
+     */
+    public void resume() {
+        workflowLifecycle.resume();
+    }
+
+    /**
+     * Returns true if the nFlow engine is currently paused.
+     */
+    public boolean isPaused() {
+        return workflowLifecycle.isPaused();
+    }
+
+    /**
+     * Returns true if the nFlow engine is currently running.
+     */
+    public boolean isRunning() {
+        return workflowLifecycle.isRunning();
+    }
+
+    /**
+     * Shuts down the nFlow engine.
```

Comment:
How? Gracefully, and returns when done? Or just triggers the shutdown and returns immediately?

Reply:
Updated javadocs